projects/components/facet/bootstrap/facet-filters/facet-filters.ts

```diff
@@ -130,4 +99,69 @@ export class BsFacetFilters implements OnChanges {
         return this.facetService.hasData(facet.aggregation, this.results);
     }
 
+    private addFacetMenu() {
+        let outFacets: Action[] = [];
+
+        outFacets.push(new Action({
+            name: `add_remove_all`,
+            text: this.userFacets.length < this.facets.length ? "msg#facet.filters.addAll" : "msg#facet.filters.removeAll",
+            icon: this.hasFacetSelected ? 
+                    (this.userFacets.length < this.facets.length ? "far fa-minus-square mr-1" : "far fa-check-square mr-1")
+                    : "far fa-square mr-1",
+            title: this.userFacets.length < this.facets.length ? "msg#facet.filters.addAll" : "msg#facet.filters.removeAll",
+            action: () => {
+                if (this.hasFacetSelected && this.userFacets.length === this.facets.length) this.facetService.removeAllFacet();
+                else this.facetService.addAllFacet();
+                this.buildFilters();
+            }
+        }));
+
+        for (let facet of this.facets) {
+            outFacets.push(new Action({
+                name: `add_remove_${facet.name}`,
+                text: facet.title,
+                icon: facet.icon,
+                selected: !!this.userFacets?.find(userFacet => userFacet.name === facet.name),
+                title: !!this.userFacets?.find(userFacet => userFacet.name === facet.name) ? "msg#facet.filters.add" : "msg#facet.filters.remove",
+                action: () => {
+                    if (this.userFacets?.find(userFacet => userFacet.name === facet.name)) this.facetService.removeFacet({name: facet.name, position: 0, hidden: false, expanded: true, view: ""})
+                    else this.facetService.addFacet({name: facet.name, position: 0, hidden: false, expanded: true, view: ""});
+                    this.buildFilters();
+                }
+            }));
+        }
+
+        let add_action = new Action({
+            name: "facets_config",
+            icon: "fas fa-cog",
+            title: "msg#facet.filters.customizeFacets",
+            children: outFacets
+        });
+        this.filters = [add_action, ...this.filters]; 
+    }
+
+    get filteredFacets() {
+        if (!this.enableCustomization) return this.facets;
+        let new_facets: FacetConfig[] = [];
+
+        if (this.userFacets) {
```

Comment:
It looks good to me, but just to be sure: For a user who opens the app for the 1st time, the idea is that he will see the list of facets injected in the facet service (or manually defined with `setDefaultFacets()`, right?)

Reply:
I'm looking at this code again, and I have a couple new remarks:
- When enableCustomization=true, you need to inject the list of facets into the service (allFacets and defaultFacets), but you also need to pass the list of facets as an input of this component. It's a pity that we need to do this twice (and of course, the 2 really need to be consistent). Couldn't we implement a logic where the facet list is only provided as an input to the component, and automatically takes care of setting up the service? (the default facets could also be passed as an input of the component)
- The for-loop below is built so that the order of the facets is determined by this.facets, even if you have a completely different list in this.userFacets. Is this on purpose?

Reply:
I made the changes for the first point you mentioned, because I do agree that there was something better to do.
For the second point, this is on purpose, because I want to keep the order of the facets consistent. If we added a functionality for the user to chose the order of display, this could be changed, but right now I think it is better like this.

Reply:
After seeing that my changes didn't go through (I forgot to test build on my side), I realized that it was going to bring unwanted breaking changes for people using the facet module.
To avoid this breaking change, I let the allFacets and defaultFacets as an injection, and made them public instead of protected + remove their defaults values of empty array if not passed in the `forRoot()`.

Reply:
Fair enough! Thanks for all the improvements.